src/lib/core/prbuf.c

```diff
@@ -369,3 +376,170 @@ prbuf_iterator_next(struct prbuf_iterator *iter, struct prbuf_entry *result)
 	result->ptr = iter->current->data;
 	return 0;
 }
+
+void
+prbuf_reader_close(struct prbuf_reader_iterator *iter)
+{
+	ibuf_destroy(&iter->buf);
+}
+
+/**
+ * Check whether buffer header is sane.
+ */
+static int
+prbuf_check_header(struct prbuf_header *header)
+{
+	if (header->version > prbuf_version) {
+		diag_set(IllegalParams,
+			 "unknown format version %d", header->version);
+		return -1;
+	}
+
+	/* Check we can the read first record and end position is correct. */
+	if (prbuf_header_size + header->begin +
+				record_size_overhead >= header->size ||
```

Comment:
The old check (`prbuf_check`) doesn't add `record_size_overhead`.

Reply:
I guess it was not correct because in `prbuf_next_record` record size for the record is accessed. This can be read outside of the buffer.